### PR TITLE
Implement csr::CSRFile

### LINKED ISSUE #11: Implement csr::CSRFile
* Need to implement Control and Status Registers file

### PR DESCRIPTION
**Pull request by VasiliyMatr**
- Implement csr::CSRFile
- Implement csr module code generation
- A few additions to common module

### CODE REVIEW

File: csr/include/sim/csr.hpp [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    if (raw_priv_level == HYPERVISOR_CODE) {
+        return PrivLevel::MACHINE;
+    }
+
+    return PrivLevel{raw_priv_level};
+}
+
+struct CSRFile final {
+    enum class AccessStatus { OK, CSR_NOT_SUPPORTED };
```
Comment (TheRedHotHabanero) [inline csr/include/sim/csr.hpp:40]: please add enum class directly to private or public part

File: csr/include/sim/csr.hpp [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  private:
+    CSRValue<XLen::XLEN_64, CSRIdx::SATP> m_satp64{};
+
+  public:
+    template <XLen xlen>
+    NODISCARD AccessStatus read(CSRIdx idx,
+                                RawCSRValue<xlen> &dst) const noexcept {
+        switch (idx) {
+        case CSRIdx::SATP:
```
Comment (TheRedHotHabanero) [inline csr/include/sim/csr.hpp:50]: how many cases will we have? if about 2-3, it`s better to use if-else statements

File: csr/src/csr.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    return m_satp64;
+}
+
+template <> void CSRFile::set(const SATP64 &value) noexcept {
+    m_satp64 = value;
+}
+
+template<>
+NODISCARD CSRFile::AccessStatus CSRFile::write<XLen::XLEN_64>(CSRIdx idx, RawCSRValue<XLen::XLEN_64> value) noexcept {
+    switch (idx) {
```
Comment (TheRedHotHabanero) [inline csr/src/csr.cpp:15]: the same as above in read function

**Review by TheRedHotHabanero — APPROVED** [review]
Approved, but please take a look at my comments

File: csr/include/sim/csr.hpp (hunk 1 above)
Comment (VasiliyMatr) [reply csr/include/sim/csr.hpp:40]: public access specifier is omitted in accordance with project codestyle

File: csr/include/sim/csr.hpp (hunk 2 above)
Comment (VasiliyMatr) [reply csr/include/sim/csr.hpp:50]: We will have more cases

File: csr/src/csr.cpp (hunk 3 above)
Comment (VasiliyMatr) [reply csr/src/csr.cpp:15]: We will have more cases

**Pull request merged**